### ISSUE TITLE
Strange result when subsetting with %in% using a set of values with duplicates

### ISSUE THREAD

**Issue by yvanrichard** [Rdatatable/data.table#1191]
The behaviour when subsetting with %in% in case of duplicates just bit me badly, with the results being very counter-intuitive.

```r
## Example data: 
dt <- data.table(a=letters[1:8], b = c(1,2,3,1,1,4,5,6))
```

I expect
```r
 dt[b %in% c(1,1,2)]
```
to return the same result as 
```r
 dt[b %in% c(1,2)]
```

but it returns
```
   a b
1: a 1
2: a 1
3: b 2
4: d 1
5: d 1
6: e 1
7: e 1
```
instead of
```
   a b
1: a 1
2: b 2
3: d 1
4: e 1
```

Is this a bug or a feature?... 

----
```
sessionInfo()
R version 3.2.1 (2015-06-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 14.04.2 LTS

locale:
 [1] LC_CTYPE=en_NZ.UTF-8       LC_NUMERIC=C               LC_TIME=en_NZ.UTF-8       
 [4] LC_COLLATE=en_NZ.UTF-8     LC_MONETARY=en_NZ.UTF-8    LC_MESSAGES=en_NZ.UTF-8   
 [7] LC_PAPER=en_NZ.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
[10] LC_TELEPHONE=C             LC_MEASUREMENT=en_NZ.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] RPostgreSQL_0.4  DBI_0.3.1        reshape2_1.4.1   data.table_1.9.4 yrpkg_2.77      

loaded via a namespace (and not attached):
[1] compiler_3.2.1 magrittr_1.5   plyr_1.8.2     tools_3.2.1    Rcpp_0.11.6    stringi_0.5-2  stringr_1.0.0 
[8] chron_2.3-46  
```

**Comment by mattdowle** [comment]
Agreed.  I believe v1.9.5 fixes that.  See README.md.   Please close if so.  It was to do with automatic indexing.  Working on release to CRAN now.

**Comment by yvanrichard** [comment]
Yes indeed, it is fixed in v1.9.5. Thank you. I hope it will come to CRAN soon.
